pkg/bootkube/parse.go

```diff
@@ -0,0 +1,78 @@
+package bootkube
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	"github.com/kubernetes-incubator/bootkube/pkg/asset"
+	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/apis/extensions/v1beta1"
+)
+
+const (
+	apiServerContainerName         = "kube-apiserver"
+	controllerManagerContainerName = "kube-controller-manager"
+)
+
+// detectServiceCIDR deserializes the '--service-cluster-ip-range' from the
+// kube-apiserver manifest
+func detectServiceCIDR(config Config) (string, error) {
+	b, err := ioutil.ReadFile(filepath.Join(config.AssetDir, asset.AssetPathAPIServer))
+	if err != nil {
+		return "", err
+	}
+	var apiServer v1beta1.DaemonSet
+	err = yaml.Unmarshal(b, &apiServer)
+	if err != nil {
+		return "", err
+	}
+
+	containers := map[string]v1.Container{}
+	for _, container := range apiServer.Spec.Template.Spec.Containers {
+		containers[container.Name] = container
+	}
+
+	if container, exists := containers[apiServerContainerName]; exists {
+		return findFlag("--service-cluster-ip-range", container.Command), nil
```

Comment:
I think we should also minimally check that we're getting a valid cidr here (net.ParseCIDR() probably enough. Otherwise --cluster-cidr with no value gets through fine.